### DISCUSSION TITLE
Compare Table Diff Pane

### DISCUSSION THREAD

**Discussion by dbrevetti** [WinMerge/winmerge#1028] [Ideas]
Right now entire row is highlighted to indicate a difference in the Diff Pane.  Would it be possible to only highlight the cells that have differences?  For instance see attached image.   Would be nice if only column D were highlighted as this is the only difference between the files/tables for this record.

![image](https://user-images.githubusercontent.com/26571000/139139566-14af3719-3864-4e95-9d70-b7bf43de369a.png)

**Comment by dbrevetti** [comment]
Also for Compare Table, would be nice if the Options menu allowed you to ignore double quotes and the delimiter (which in this case is a comma since it is a csv file)

**Comment by sdottaka** [comment]
WinMerge can normally highlight the differences between characters in a cell, as shown below.

![image](https://user-images.githubusercontent.com/98126/139160209-cafe752e-b569-45aa-911f-5c86f082986f.png)

However, WinMerge stops calculating inline differences in very long lines, as this slows down the program. Do you have very long lines in your files?

You can use the IgnoreFieldsComma prediffer plugin to ignore differences within a specified range of column numbers, as shown below.
Using this plugin will internally reduce the length of the lines to be compared. Therefore, if you try to specify a column range that should obviously match, it may be highlighted.

![image](https://user-images.githubusercontent.com/98126/139160545-a405fc12-b824-4cdc-944d-bd7876210901.png)

**Comment by dbrevetti** [comment]
Thanks for the great info.  I didn't know about the line length although the lines aren't what I would consider long.  Header is 350 characters but the data lines are just 175 or less.  

Would be nice if you had an option to turn off the  "stop calculating inline differences on long lines" so if user is willing to wait they could see them.

Thanks for the info on the plugins.  I'll check it out.

**Comment by sdottaka** [comment]
> Header is 350 characters but the data lines are just 175 or less.

Hmm. It doesn't seem to be that long. Is it possible to upload those files?

**Comment by grovduck** [comment]
I think I'm running into this same issue (and I'm sure I'm doing something wrong).  I'm using two toy files with these contents:

    A,B,C
    1,2,3

    A,B,C
    1,2,4

And here is my result.  I'm expecting that only field C should be highlighted.

![image](https://user-images.githubusercontent.com/1706997/198362952-c9a0a225-6a03-43a7-a8ea-0be0939d82ee.png)

I've set the prediffer to ignore columns 7-1000 as suggested.  Any help would be much appreciated.

**Comment by sdottaka** [comment]
Perhaps you can get the expected result by setting as follows in the Options dialog.

![image](https://user-images.githubusercontent.com/98126/198410595-c2937f90-17bd-4819-bf63-29ba17295416.png)

**Reply by grovduck** [reply]
@sdottaka - Thanks so much, this solved it for me.  Sorry for the noise
